code/modules/mob/living/damage_procs.dm

```diff
@@ -184,9 +184,12 @@
 /mob/living/proc/adjustOxyLoss(amount, updating_health = TRUE, forced = FALSE, required_biotype, required_respiration_type = ALL)
 	if(!forced && (status_flags & GODMODE))
 		return
-	if(iscarbon(src))
-		var/obj/item/organ/internal/lungs/affected_lungs = getorganslot(ORGAN_SLOT_LUNGS)
-		if(!forced && !(affected_lungs?.respiration_type & required_respiration_type))
+	var/obj/item/organ/internal/lungs/affected_lungs = getorganslot(ORGAN_SLOT_LUNGS)
+	if(!affected_lungs)
+		if(!forced && !(mob_respiration_type & required_respiration_type))  // if the mob has no lungs, use mob_respiration_type
+			return
+	else 
+		if(!forced && !(affected_lungs.respiration_type & required_respiration_type)) // otherwise use the lungs' respiration_type
 			return
```

Comment:
```suggestion
	if(!forced && !affected_lungs)
		if(!(mob_respiration_type & required_respiration_type))  // if the mob has no lungs, use mob_respiration_type
			return
	else 
		if(!(affected_lungs.respiration_type & required_respiration_type)) // otherwise use the lungs' respiration_type
			return
```

If it's forced, we don't even need to bother with these checks.

I'm also just now seeing that this is pretty much a duplicate check to what you wrote in `setOxyLoss()`. Why not spin it out into it's own proc that returns TRUE/FALSE to reduce copypasting, like:

```
/mob/living/proc/adjustOxyLoss(..., required_respiration_type)
...
if(!forced && !check_respiration_viability(required_respiration_type))
    return    

/// dmdoc here, e.g. what the return values are and mean
/mob/living/proc/check_respiration_viability(required_respiration_type)
    var/obj/item/organ/internal/lungs/affected_lungs = getorganslot(ORGAN_SLOT_LUNGS)
    if(!affected_lungs)
	if(!(mob_respiration_type & required_respiration_type))
		return FALSE
    else 
	if(!(affected_lungs.respiration_type & required_respiration_type))
		return FALSE
         
    return TRUE
```

This is just an example, would work a lot better to have the similar behaviors in one unified proc rather than risk having an odd desynchronization in behaviors (like a review I include later).

Reply:
yeah there's no need no more